addons/mrp/models/mrp_production.py

```diff
@@ -520,6 +520,12 @@ def write(self, vals):
                     raise UserError(_('You cannot move a planned manufacturing order.'))
             if 'move_raw_ids' in vals and production.state != 'draft':
                 production.move_raw_ids.filtered(lambda m: m.state == 'draft')._action_confirm()
+            if production.state != 'draft':
```

Comment:
Could you use the special state domain instead. We use it already in `stock_picking.py` for instance: 
https://github.com/odoo/odoo/blob/20c3cd317de35a765fb0be80aa3b331d51cf708a/addons/stock/models/stock_picking.py#L334-L337

Reply:
I didn't know about that, good to know :+1: